### ISSUE TITLE
Exclusive Lock Error with Vagrant NFS Mounts

### ISSUE THREAD

**Issue by PabloKowalczyk** [FriendsOfPHP/PHP-CS-Fixer#3460]
#### The PHP version you are using (`$ php -v`):
=> `PHP 7.2.1-1+ubuntu16.04.1+deb.sury.org+1 (cli) (built: Jan  5 2018 13:54:13) ( NTS )`

#### PHP CS Fixer version you are using (`$ php-cs-fixer -V`):
=> `PHP CS Fixer 2.10.0 Bowling Bear`

#### The command you use to run PHP CS Fixer:
=> `vendor/bin/php-cs-fixer fix`

Hello, i encountered some issues with Vagrant NFS.

There is a problem with `\PhpCsFixer\Cache\FileHandler::write()` which uses `file_put_contents` (in line 86)  with third parameter `LOCK_EX` - this causes error on NFS Vagrant mounts:
```
[Symfony\Component\Filesystem\Exception\IOException]
Failed to write file ".php_cs.cache", "no reason available".
```
The same problem was resolved here https://github.com/symfony/symfony/issues/25336  by simply remove `LOCK_EX`.

**Comment by SpacePossum** [comment]
Thanks for reporting!
Related to this https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/1854 issue, which as not resolved as it was hard to reproduce.
Could you propose a fix, or if a PR was raised test it to see if it works on a setup that is currently broken?

**Comment by SpacePossum** [comment]
Labelled `Easy Pick`; to resolve this issue the `LOCK_EX` flag has to be removed when handling the cache file. The target branch should be the lowest we support.

**Comment by chrisandrewcl** [comment]
Just in case someone lands here from Google: I lost some considerable time trying to figure this out. I have tried several different configurations. Nothing worked, so I stepped down a level and tried to configure it directly between the guest instance and the host machine (i.e. manually editing `/etc/exports` and issuing `mount`s). It paid off :)

The final minimal configuration that worked for my setup (Ubuntu 17.10, Vagrant 2.0.2) is this:

``` 
config.vm.synced_folder ".", "/vagrant", 
    :nfs => true, 
    :nfs_version => 4, 
    :nfs_udp => false, 
    :mount_options => ['sec=sys']
```
